specification/cognitiveservices/OpenAI.Inference/models/chat.completions.tsp

```diff
@@ -49,6 +54,16 @@ model ChatMessage {
   @added(ServiceApiVersions.v2023_07_01_Preview)
   @projectedName("json", "function_call")
   functionCall?: FunctionCall;
+
+  @doc("""
+    Additional context data associated with a chat message when requesting chat completions using compatible Azure
+    OpenAI chat extensions. This includes information like the intermediate data source retrievals used to service a
+    request.
+    This context information is only populated when using Azure OpenAI with chat extensions capabilities configured.
+  """)
+  @added(ServiceApiVersions.v2023_08_01_Preview)
+  @projectedName("csharp", "AzureExtensionsContext")
+  context?: AzureChatExtensionsMessageContext;
```

Comment:
I believe `context` appears only once in the top-level message and shouldn't appear again inside any message in the context itself. Should the type be updated to precisely reflect this behavior? For example, introducing:

`model AzureChatMessage { ..., context: AzureChatExtensionsMessageContext }`

Reply:
I agree. Since `AzureChatExtensionMessageContext` is just  a wrapper of a list of `ChatMessage`s , it makes the definition look oddly circular. 

However, if I understand the swagger correctly, the definition is in fact circular, judging by how `context is [defined here](https://github.com/Azure/azure-rest-api-specs-pr/pull/13558/files#diff-6e26583b8f5643469b0ed2446d678032411ad9bcf78200c16e91846f7976a9a9R1231), where `#/component/schemas/message` corresponds to [this definition](https://github.com/Azure/azure-rest-api-specs-pr/pull/13558/files#diff-6e26583b8f5643469b0ed2446d678032411ad9bcf78200c16e91846f7976a9a9R1190)

Reply:
Technically, a circular type is still a valid type to use but it is not the most accurate given that the context should appear in the root node only. My point is that we can create a more accurate representation here.